### PR TITLE
Add endpoint to upgrade all node deployments within the cluster

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: Adds endpoint to upgrade all node deployments within the cluster.

**Documentation**: n/a

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/upgrade.go [hunk 1]
```diff
@@ -91,6 +94,78 @@ func getClusterNodeUpgrades(updateManager UpdateManager, projectProvider provide
...
+
+		machineDeployments, err := machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).List(metav1.ListOptions{})
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		var updateErrors []error
+		for _, machineDeployment := range machineDeployments.Items {
+			machineDeployment.Spec.Template.Spec.Versions.Kubelet = req.Body.Version.String()
+			_, err = machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).Update(&machineDeployment)
```
Comment (p0lyn0mial) [inline api/pkg/handler/upgrade.go:165]: before updating we must check if the kubelet version is compatible with the cluster version (use `ensureVersionCompatible` method ?) Would be nice if we could cover this scenario with a unit test.

File: api/pkg/handler/upgrade.go [hunk 2]
```diff
@@ -91,6 +94,78 @@ func getClusterNodeUpgrades(updateManager UpdateManager, projectProvider provide
...
+			if err != nil {
+				updateErrors = append(updateErrors, err)
+			}
+		}
+
+		if len(updateErrors) > 0 {
+			return nil, fmt.Errorf("failed to update some node deployments: %+q", updateErrors)
```
Comment (p0lyn0mial) [inline api/pkg/handler/upgrade.go]: how does the client parse the response ? wouldn't be better to return human readable message along with a list of additional errors ? e.g. `{"error": "failed to update some node deployments", "details": ["first error", "second error"]"`

**Review by p0lyn0mial — CHANGES_REQUESTED** [review]

File: api/pkg/handler/upgrade.go (hunk 2 above)
Comment (maciaszczykm) [reply api/pkg/handler/upgrade.go]: It is parsed like every other error message. Sample notification will show `Error 500: failed to update some node deployments: ["err1", "err2"]`.

File: api/pkg/handler/upgrade.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/handler/upgrade.go:165]: Yes, sure it should be done. I will do it.

File: api/pkg/handler/upgrade.go (hunk 2 above)
Comment (maciaszczykm) [reply api/pkg/handler/upgrade.go]: Done.

File: api/pkg/handler/upgrade.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/handler/upgrade.go:165]: Done.

**Comment by maciaszczykm** [comment]
@p0lyn0mial PTAL

**Pull request merged**